File: src/cpp/two-sum.cpp

// Author - Shadowpii

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp; 
        vector<int> ans;
        for (int i = 0; i < nums.size(); i++)
            if (mp.count(target - nums[i])) 
            {
                ans.push_back(mp[target - nums[i]]);
                ans.push_back(i);
                return ans;
            }
            else 
                mp[nums[i]] = i;
        return ans;
    }
};

File: src/java/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int numsSize = nums.length - 1;
        int[] results = null;
        for (int i = 0; i <= numsSize; i++) {
            for(int j = i+1; j <= numsSize; j++) {
                if (nums[i]+nums[j] == target) {
                    results = new int[]{i, j};
                }
            }
        }
        return results;
    }
}